lib/dbservice_web/controllers/user_controller.ex

```diff
@@ -201,14 +201,42 @@ defmodule DbserviceWeb.UserController do
   defp get_group(group, quiz_flag) do
     child_id = group.child_id
     batch = Batches.get_batch!(child_id)
+    class_batch_id = batch.batch_id
     quiz_id = batch.parent_id
     quiz_group = Groups.get_group_by_child_id(quiz_id)
     quiz_group_id = quiz_group.id
 
     group_id = if quiz_flag, do: quiz_group_id, else: group.id
     group_sessions = GroupSessions.get_group_session_by_group_id(group_id)
 
-    Enum.map(group_sessions, &get_group_session(&1))
+    filtered_sessions = filter_sessions(group_sessions, quiz_flag, class_batch_id)
+
+    Enum.map(filtered_sessions, &get_group_session/1)
+  end
+
+  defp filter_sessions(group_sessions, quiz_flag, class_batch_id) do
+    group_sessions
+    |> Enum.filter(&session_filter(&1, quiz_flag, class_batch_id))
+  end
+
+  defp session_filter(group_session, quiz_flag, class_batch_id) do
+    session = get_group_session(group_session)
+
+    case session do
+      nil ->
+        false
+
+      _ ->
+        if quiz_flag do
+          if !is_nil(session.meta_data["batch_id"]) do
+            session.meta_data["batch_id"] == class_batch_id and session.platform == "quiz"
+          else
+            false
+          end
+        else
+          session.platform == "meet"
```

Comment:
why this? you can just put true maybe? if quiz flag is false then eveyr other platform is allowed. not just meet

Reply:
But for live classes, shouldn't the platform be always meet?

Reply:
In Gurukul, we show only Live classes and Quiz. My intent was to show "meet" sessions for live classes and "quiz" sessions for Tests.